src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+import org.example.bookstore.model.User;
+import org.example.bookstore.repository.OrderItemRepository;
+import org.example.bookstore.repository.OrderRepository;
+import org.example.bookstore.repository.ShoppingCartRepository;
+import org.example.bookstore.service.OrderService;
+import org.example.bookstore.service.ShoppingCartService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartService shoppingCartService;
+
+    @Override
+    public OrderDto createOrder(Authentication authentication, CreateOrderRequestDto requestDto) {
+        User user = findUser(authentication);
+        ShoppingCart cart = shoppingCartRepository.findByUserId(user.getId());
+        if (cart.getCartItems().isEmpty()) {
+            throw new EntityNotFoundException("Can't create an order from an empty cart!");
+        }
+        Order order = orderMapper.toModel(requestDto);
+        order.setUser(user);
+        order.setOrderDate(LocalDateTime.now());
+        order.setOrderItems(
+                cart.getCartItems().stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = new OrderItem();
+                    orderItem.setOrder(order);
+                    orderItem.setBook(cartItem.getBook());
+                    orderItem.setQuantity(cartItem.getQuantity());
+                    orderItem.setPrice(cartItem.getBook().getPrice());
+                    return orderItem;
+                })
+                .collect(Collectors.toSet())
+        );
+        order.setTotal(
+                order.getOrderItems().stream()
+                .map(item ->
+                    item.getPrice().multiply(BigDecimal.valueOf(item.getQuantity()))
+                )
+                .reduce(BigDecimal.ZERO, BigDecimal::add)
+        );
+        shoppingCartService.clearShoppingCart(cart);
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public Page<OrderDto> getOrders(Authentication authentication, Pageable pageable) {
+        return orderRepository.findAllByUserId(findUser(authentication), pageable)
+            .map(orderMapper::toDto);
+    }
+
+    @Override
+    public Page<OrderItemDto> getOrderItems(Authentication authentication,
+                                            Long orderId,
+                                            Pageable pageable
+    ) {
```

Comment:
```suggestion
    public Page<OrderItemDto> getOrderItems(
        Authentication authentication,
        Long orderId,
        Pageable pageable
    ) {
```
I prefer this style. You can use it if you like it